VC4 GPU driver for Raspberry Pi & DisplayLink USB monitors

xranby opened the pull request:
Using Eric Anholts MIT-licensed Mesa and kernel DRM driver
for the 2708 (aka the 2835), the chip that's in the Raspberry Pi.
http://anholt.livejournal.com/
http://dri.freedesktop.org/wiki/VC4/

x11r7/xserver_xorg-server: 1.16.99.901 first RC for X server 1.17 …
x11r7/xlib_xtrans: 1.3.5
- The modesetting driver has been merged into the server code base,
  simplifying ongoing maintenance by coupling it to the X server
  ABI/API release schedule. This now includes DRI2 support (so that GLX
  works correctly) along with Glamor support (which handles DRI3)
  http://lists.x.org/archives/xorg-announce/2014-October/002491.html

The Linux kernel has two drivers for use by DisplayLink USB monitors
The udlfb framebuffer driver to be used with xdriver_xf86-video-udlfb
The udl KMS driver to be used with the modesetting driver.
The modesetting driver also support the VC4 GPU using the Linux VC4 DRM driver.

xranby committed:
Add xdriver_xf86-video-modesetting & xdriver_xf86-video-udlfb
The Linux kernel has two drivers for use by DisplayLink USB monitors
The udlfb framebuffer driver to be used with xdriver_xf86-video-udlfb
The udl KMS driver to be used with xdriver_xf86-video-modesetting

Signed-off-by: Xerxes Rånby <xerxes@gudinna.com>

xranby committed:
mesa3d: Gallium VC4 GPU driver for Raspberry Pi.

Using Eric Anholts MIT-licensed Mesa and kernel DRM driver
for the 2708 (aka the 2835), the chip that's in the Raspberry Pi.
http://anholt.livejournal.com/
http://dri.freedesktop.org/wiki/VC4/

Signed-off-by: Xerxes Rånby <xerxes@gudinna.com>

xranby committed:
x11r7/xserver_xorg-server: 1.16.99.901 first RC for X server 1.17
x11r7/xlib_xtrans: 1.3.5

* The modesetting driver has been merged into the server code base,
  simplifying ongoing maintenance by coupling it to the X server
  ABI/API release schedule. This now includes DRI2 support (so that GLX
  works correctly) along with Glamor support (which handles DRI3)
  http://lists.x.org/archives/xorg-announce/2014-October/002491.html

x11r7/xdriver_xf86-video-modesetting: Removed.

Signed-off-by: Xerxes Rånby <xerxes@gudinna.com>

xranby committed:
mesa3d: Refresh mesa3d/mesa3d-0001-Fix-khrplatform.h-not-installed-if-EGL-is-disabled.patch

Signed-off-by: Xerxes Rånby <xerxes@gudinna.com>

xranby committed:
x11r7/xdriver_xf86-video-udlfb: Make compatible with xorg-server 1.16.99.901

Signed-off-by: Xerxes Rånby <xerxes@gudinna.com>

xranby committed:
x11r7/xserver_xorg-server: --enable-glamor & libepoxy: glamor build dependency

	* enable Glamor support (which handles DRI3) for the
	  modesetting driver.

	  http://www.freedesktop.org/wiki/Software/Glamor/

Signed-off-by: Xerxes Rånby <xerxes@gudinna.com>

gamaral commented:
Sorry for the delay, I've been pretty busy. I'll give it a try this thanksgiving weekend and pull if all goes well. :)

xranby commented:
I recommend that we wait merging this.
The three core projects llinux kernel, mesa3d and xorg are changing rapidly but will be in a consumable state in January 2015. I think that is when all pieces fall into place.

current status:
mesa3d is expected to be released in a month from now & vc4 work is evolving fast ->
http://cgit.freedesktop.org/mesa/mesa/log/?qt=grep&q=vc4
linux vc4 kernel branch evolve fast ->
https://github.com/anholt/linux/commits/vc4
and xorg 1.17 have a release candidate with the glamor modesetting driver that i consider "stable".

I will try merge this work into buildroot master directly when there is upstream releases of linux kernel, xorg and mesa3d got stabel variants of the raspberry pi vc4 drivers included.

xranby committed:
mesa3d: enable-gallium-gbm enable-gallium-egl use 2014-11-24 git

Signed-off-by: Xerxes Rånby <xerxes@gudinna.com>

xranby committed:
openal: OpenAL Soft is an LGPL, implementation of the OpenAL 3D audio API.

It's forked from the open-sourced Windows version available originally from
openal.org's SVN repository (now defunct).
OpenAL provides capabilities for playing audio in a virtual 3D environment.

Signed-off-by: Xerxes Rånby <xerxes@gudinna.com>

xranby committed:
xdotool: fake keyboard/mouse input, window management, and more

Signed-off-by: Xerxes Rånby <xerxes@gudinna.com>

xranby committed:
mesa3d: use 2014-12-06 git

Signed-off-by: Xerxes Rånby <xerxes@gudinna.com>
